tests/input.spec.ts

```diff
@@ -0,0 +1,210 @@
+import { test, expect } from '@playwright/test';
+import { hexToRGB, getCustomEventParam, waitForCustomEventWithParam } from './lib/helper.ts';
+
+test.beforeEach(async ({ page }) => {
+  await page.goto('/input-test.html');
+});
+
+test('Given type text, renders text input', async ({ page }) => {
+  const inputText = await page.getByTestId('input-text')
+  await expect(inputText).toBeVisible()
+  await expect(inputText.locator('input').first()).toHaveAttribute('type', 'text')
+});
+
+test('Given no type, renders text input', async ({ page }) => {
+  const inputDefault = await page.getByTestId('input-default')
+  await expect(inputDefault).toBeVisible()
+  await expect(inputDefault.locator('input').first()).toHaveAttribute('type', 'text')
+});
+
+test('Given value attribute, initializes with given value as input value', async ({ page }) => {
+  const inputText = await page.getByTestId('input-text')
+  await expect(inputText).toBeVisible()
+  await expect(inputText.locator('input').first()).toHaveValue('value')
+});
+
+test('Displays "*" next to the label if required', async ({ page }) => {
+  const inputRequired = await page.getByTestId('input-required')
+  await expect(inputRequired).toBeVisible()
+  await expect(inputRequired.locator('input[required]').first()).toBeVisible()
+  const asterisk = await inputRequired.getByText(">:)").evaluate((e1) => {     
+    return {
+      'content': window.getComputedStyle(e1,':after').getPropertyValue('content'),
+      'color': window.getComputedStyle(e1, ':after').color,
+    }
+  })
+  expect(asterisk.content).toContain("*")
+  expect(asterisk.color).toBe(hexToRGB('danger-fg'))
+});
+
+test('Displays label on top of input by default', async ({ page }) => {
+  const inputDefault = await page.getByTestId('input-default')
+  await expect(inputDefault.getByText('default')).toBeVisible()
+  await expect(await inputDefault.locator("input:below(:text('default'))").first()).toBeVisible()
+});
+
+test('Displays label on top of input with labelposition top', async ({ page }) => {
+  const inputLabelTop = await page.getByTestId('input-label-top')
+  await expect(inputLabelTop.getByText(':D')).toBeVisible()
+  await expect(await inputLabelTop.locator("input:below(:text(':D'))").first()).toBeVisible()
+});
+
+test('Displays label to left of input with labelposition left', async ({ page }) => {
+  const inputLabelLeft = await page.getByTestId('input-label-left')
+  await expect(inputLabelLeft.getByText('D:')).toBeVisible()
+  await expect(await inputLabelLeft.locator("input:right-of(:text('D:'))").first()).toBeVisible()
+});
+
+test('Given type number, initializes with given number value', async ({ page }) => {
+  const inputNumber = await page.getByTestId('input-number')
+  await expect(inputNumber).toBeVisible()
+  await expect(inputNumber.locator('input').first()).toHaveValue('3.14159')
+});
+
```

Comment:
did not include testing for type=`integer`, it's not included in prime stories but i saw it in the code. should i add?

Reply:
added

Reply:
Ah yep, we should add type=integer to the stories as well. Not a blocker though